tests/i18n/tests.py

```diff
@@ -1860,6 +1860,15 @@ def test_page_with_dash(self):
         response = self.client.get('/de-simple-page/')
         self.assertEqual(response.content, b'Yes')
 
+    def test_no_redirect_on_404(self):
+        """
+        Request for non-existent url can cause an unneeded redirect to
```

Comment:
"nonexistent" is one word. Suggested wording:
```
Request for a nonexistent URL shouldn't cause a redirect to
/<defaut_language>/<request_url> when prefix_default_language=False
and /<default_language>/<request_url> has a URL match (#27402).
```
Can you add a check that the requested URL matches one of the patterns? Right now if the new urlpattern is inadvertently removed, the test still passes but doesn't work as a regression test.